common/scripts/x86_64/Ubuntu_16.04/remote/unseal_vault.sh

```diff
@@ -1,33 +1,132 @@
 #! /bin/bash
 set -e
 
-export ADMIRAL_ENV="/etc/shippable/admiral.env"
-export BOOT_WAIT=10
-export VAULT_ADDR='http://127.0.0.1:8200'
+readonly VAULT_API_RESPONSE_FILE="/tmp/.vault_response"
+readonly ADMIRAL_ENV="/etc/shippable/admiral.env"
+readonly BOOT_WAIT=10
+
+##################### Begin Vault adapter ######################################
```

Comment:
probably move vault adapter to a common file, so that it can be used in multiple places.

Reply:
this wasn't moved in a different file because this script is supposed to be a standalone bash executable that can unseal vault. breaking it down means systemd now needs to import the common file from some location, then import the rest of the script from some other location, combine the two and then execute.
also, since this script will be copied over to remote hosts(for multi-box installs), its better to keep all the code in one place rather than breaking it down and then assembling the pieces when required.
we can discuss other approach if this still doesnt seem convincing

Reply:
hmm ok, but this will be repeated in other OS' `unseal_vault.sh` files. So, I thought if this could be abstracted out and made into a adapter this could be better. But, I guess we need the extra step of copying the adapter to `/etc/vault.d` as well to run the unseal_vault.sh file and source it in this file. Maybe we should rationalize the scripts at a later date,?

Reply:
ya, it'll be repeated in other OS's to. we'll need to rethink how scripts are bundled if we want to remove the duplicated code. e.g all upstart scripts can be in one place while systemd in one. but that is out of scope for this issue.

Reply:
ok